### DISCUSSION TITLE
How to do stuff when using as a decorator?

### DISCUSSION THREAD

**Discussion by 3cha1n** [pavdmyt/yaspin#210] [Q&A] [answered]
Can anybody tell me how to change yaspin attributes inside the function when it's used as a decorator?
E.g. so I can't do:
sp.text="New progress text.."
..because there's no 'sp', so is there any other way I can alter the spinner text when yaspin is being used as a decorator?

**Comment by pavdmyt** [comment, marked as answer]
Spinner instance is not passed inside of the decorated function as of now: https://github.com/pavdmyt/yaspin/blob/master/yaspin/core.py#L109, so it is not possible to modify spinner attributes when using it as a decorator.

Technically it should be easy to pass spinner instance inside the function via e.g. pre-defined `spinner` argument. This is the first thing that came to my mind, have to think more carefully if better API is possible for this use-case.

**Reply by 3cha1n** [reply]
Hi Pavel,
Thanks for your response!
In the meantime I'm using manual mode which seems to be perfect so far for what I need 👍🏼
I'm enjoying the tool, it's great!
Regards, 
Joe

**Reply by pavdmyt** [reply]
This was addressed in #256
